ExampleMod/Common/UI/ExampleDisplaySets/ExampleReversedBarsDisplay.cs

```diff
@@ -0,0 +1,222 @@
+
+			_hpHovered = isHovered;
+		}
+
+		public override bool PreHover(out bool hoveringLife) {
+			hoveringLife = _hpHovered;
+			return _hpHovered || _mpHovered;
+		}
+
+		public override void PreDrawResources(PlayerStatsSnapshot snapshot) {
+			_hpSegmentsCount = snapshot.AmountOfLifeHearts;
+			_mpSegmentsCount = snapshot.AmountOfManaStars;
+			_maxSegmentCount = 20;
+			_hpFruitCount = snapshot.LifeFruitCount;
+			_hpPercent = (float)snapshot.Life / snapshot.LifeMax;
+			_mpPercent = (float)snapshot.Mana / snapshot.ManaMax;
+			preparedSnapshot = snapshot;
+		}
+
+		// The methods below were copied from HorizontalBarsPlayerResourcesDisplaySet and modified to account for the changed right panels
+		private void LifePanelDrawer(int elementIndex, int firstElementIndex, int lastElementIndex, out Asset<Texture2D> sprite, out Vector2 offset, out float drawScale, out Rectangle? sourceRect) {
+			sourceRect = null;
+			offset = Vector2.Zero;
+			sprite = _panelLeft;
+			drawScale = 1f;
+			if (elementIndex == lastElementIndex) {
+				sprite = _panelRightHP;
+				// Offset swapped with offset from ManaPanelDrawer
+				//offset = new Vector2(-16f, -10f);
+				offset = new Vector2(-16f, -6f);
+			}
+			else if (elementIndex != firstElementIndex) {
+				sprite = _panelMiddleHP;
+			}
+		}
+
+		private void ManaPanelDrawer(int elementIndex, int firstElementIndex, int lastElementIndex, out Asset<Texture2D> sprite, out Vector2 offset, out float drawScale, out Rectangle? sourceRect) {
+			sourceRect = null;
+			offset = Vector2.Zero;
+			sprite = _panelLeft;
+			drawScale = 1f;
+			if (elementIndex == lastElementIndex) {
+				sprite = _panelRightMP;
+				// Offset swapped with offset from LifePanelDrawer
+				//offset = new Vector2(-16f, -6f);
+				offset = new Vector2(-16f, -10f);
+			}
+			else if (elementIndex != firstElementIndex) {
+				sprite = _panelMiddleMP;
+			}
+		}
+
+		private void LifeFillingDrawer(int elementIndex, int firstElementIndex, int lastElementIndex, out Asset<Texture2D> sprite, out Vector2 offset, out float drawScale, out Rectangle? sourceRect) {
+			sprite = _hpFill;
+			if (elementIndex >= _hpSegmentsCount - _hpFruitCount)
+				sprite = _hpFillHoney;
+
+			HorizontalBarsPlayerReosurcesDisplaySet.FillBarByValues(elementIndex, sprite, _hpSegmentsCount, _hpPercent, out offset, out drawScale, out sourceRect);
+		}
+
+		private void ManaFillingDrawer(int elementIndex, int firstElementIndex, int lastElementIndex, out Asset<Texture2D> sprite, out Vector2 offset, out float drawScale, out Rectangle? sourceRect) {
+			sprite = _mpFill;
+			HorizontalBarsPlayerReosurcesDisplaySet.FillBarByValues(elementIndex, sprite, _mpSegmentsCount, _mpPercent, out offset, out drawScale, out sourceRect);
+		}
+
+		private void LifePanelContext(int elementIndex, int firstElementIndex, int lastElementIndex, out PlayerStatsSnapshot snapshot, out IResourceDrawSource drawSource) {
+			snapshot = preparedSnapshot;
```

Comment:
Not a fan of this boilerplate. tML should handle this since it's exactly the same in every implementation